### DISCUSSION TITLE
Executing `npm ci -ws` removes the turbo command

### DISCUSSION THREAD

**Discussion by thany** [vercel/turborepo#9500] [Help]
### Summary

When I execute `npm ci -ws` just to test if all packages are properly installed, the `turbo` command doesn't get installed alongside all the other packages (that do get properly installed).

So after this command I can't do `npm run build` which on its turn calls `turbo build`:

```
'turbo' is not recognized as an internal or external command, operable program or batch file.
```
(that's the Microsoft way of saying a command doesn't exist)

I have to do an additional `npm i` in order to get turbo to work again. And I wonder why that is. Is this intentional / by design? Who's design would it be anyway, because it may just as well be a quirk of npm, not turbo. I have no idea, so I'm starting here.

I'm on Node.js 20.18 and npm 10.9.1. I should also mention that I don't have turbo installed globally; never have, never will.

I'm just starting a discussion instead of a bug report, because for one I'm not sure if it's a turbo bug or an npm bug. And secondly the bug report template demands a link to a repro repo, which is pretty well irrelevant, iyam.

Additionally, the `npm ci` command sometimes complains that it can't unlink `turbo.exe` from somewhere deep within the node_modules directory. This is because a `turbo.exe` process remains running on my pc, even though it had finished doing its thing long ago. This feels like a bug as well, that also can't be reproduced by just linking to an existing turbo repo - you need to be on Windows, for starters, and whichever repo you're on, doesn't even matter at all as long as it's a turbo repo.

**Comment by anthonyshew** [comment]
I just tried out `npm ci -ws` and ran `turbo` afterward and it ran as I'd expect. I'm on macOS, though. I expect this is an npm + Windows bug, since we don't do anything exotic with our `.exe`. We do ship a binary onto npm which is less common but we're far from the only ones doing this.

> This is because a turbo.exe process remains running on my pc, even though it had finished doing its thing long ago.

This is likely the background daemon still running (as expected). It stays on until it hits its timeout, since we don't know when you've decided you're done running `turbo` commands for the day. If you'd prefer to not use the daemon, you can[ turn it off in `turbo.json`](https://turbo.build/repo/docs/reference/configuration#daemon) or [use the `--no-daemon` flag](https://turbo.build/repo/docs/reference/run#--daemon-and---no-daemon).

**Reply by thany** [reply]
Thank you for explaining. First, I don't know why the turbo command gets lost, maybe it doesn't happen on macOS, maybe something weird happens on my pc. I'm not saying this is a fault in the turbo package, I'm just asking for help. I don't really know where else to go until I know what's going on. Ultimately I just want it so that `npm install` installs everything a fellow developer needs, without having to manually/globally install anything. Of course, this is important for a CI environment, which luckily still works fine.

About the daemon thing - makes sense. Maybe a way for it to detect when it needs to get uninstalled that it closes itself somehow. I understand that it's not super common to remove the node_modules filter entirely, or indeed do a `npm ci`, but I imagine updating the package might have a similar effect (on every OS), where the executable can't be overwritten because it's currently running.

**Reply by anthonyshew** [reply]
No worries, and we'll leave this open in case anyone else happens to come around with a great answer. 😄
